crates/bitwarden-auth/src/identity/login_via_password/password_login_error.rs

```diff
@@ -0,0 +1,423 @@
+use bitwarden_core::key_management::MasterPasswordError;
+use bitwarden_error::bitwarden_error;
+use thiserror::Error;
+
+use crate::identity::api::response::{
+    InvalidGrantError, LoginErrorApiResponse, OAuth2ErrorApiResponse, PasswordInvalidGrantError,
+};
+
+/// Errors that can occur during password-based login.
+///
+/// This enum covers errors specific to the password authentication flow, including
+/// credential validation, KDF processing, and API communication errors.
+#[bitwarden_error(flat)]
+#[derive(Debug, Error)]
+pub enum PasswordLoginError {
+    /// The username (email) or password provided was invalid.
+    ///
+    /// This error is returned by the server when:
+    /// - The email address doesn't exist in the system
+    /// - The master password hash doesn't match the stored hash
+    ///
+    /// # Note
+    /// For security reasons, the server doesn't distinguish between "user not found"
+    /// and "wrong password" to prevent user enumeration attacks.
+    #[error("Invalid username or password provided.")]
+    InvalidUsernameOrPassword,
+
+    /// Failed to derive master password authentication data from the provided password and KDF
+    /// settings.
+    ///
+    /// This error occurs during local cryptographic processing before the API call, typically when:
+    /// - The KDF parameters are invalid (e.g., iterations below minimum threshold)
+    /// - The KDF algorithm is unsupported or corrupted
+    /// - Memory allocation fails during Argon2id processing
+    ///
+    /// # Common Causes
+    /// - KDF iterations below the minimum security threshold (5000 for PBKDF2)
+    /// - Corrupted prelogin data
```

Comment:
Seems these are duplicated with the 3 lines above?

Reply:
f6d89b29bfe8b7618dfce3f5bae0676c1f092e49